src/main/java/uk/gov/companieshouse/chs/gov/uk/notify/integration/api/templatepersonalisation/TemplatePersonaliser.java

```diff
@@ -187,6 +191,9 @@ private void populateLetterWithTodaysDate(Context context,
             }
             context.setVariable(TODAYS_DATE, date);
         }
+        if(CSIDVDEFLET.equals(templateLookupKey)) {
+            context.setVariable(ACTION_DUE_DATE, LocalDate.now().plusDays(28).format(DATE_FORMATTER));
```

Comment:
We should enable support for letter regeneration by adding the 28 days to either today's date or the original date the letter was sent, we could add this logic to the if statement right above this

More info on regenerating letters [here](https://companieshouse.atlassian.net/wiki/spaces/IDV/pages/5585699109/How+to+view+letters+previously+sent+to+PSCs)